pkg/core/fs_ro_ops.go

```diff
@@ -184,34 +186,59 @@ func (fs *readOnlyFsInternal) ReleaseFileHandle(
 	return
 }
 
-func isDir(fsEntry *fsEntry) bool {
-	return fsEntry.hash != ""
+func (fs *readOnlyFsInternal) FlushFile(
+	ctx context.Context,
+	op *fuseops.FlushFileOp) (err error) {
+	// noop
+	return
+}
+
+type fsNodeToAdd struct {
+	parentINode fuseops.InodeID
+	FsEntry     FsEntry
+}
+
+// FsEntry is a node in the filesystem
+type FsEntry struct {
```

Comment:
keep this type private?

Reply:
no.
Since we expose `Bundle.ReadAt()` we need to expose the types of its parameters as well. Hence `FsEntry` and not `fsEntry`.
Aside: I'd really like to move all this to some `pkg/core/fuse` or `pkg/fuse` package. This would probably need to create a common testing utility pkg since testing fuse bends on a lot of the bundle testing code.

Reply:
follow-up note: disagree - won't fix

Reply:
> I'd really like to move all this to some `pkg/fuse` package.

creating a package for the filesystem implementation stuff has been a todo for a while, although idk if we have an explicit issue for it.

separating out the tests, specifically, would be much of the time cost.

Reply:
Yes agree. Moving this to plg/fuse is going to be the next move.
This way we remove a lot of options from bundle, remove the bumdle.GetAt and move back FsEntry to unexported